src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5309,15 +5309,17 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
             foreach2(args, tparams) { (arg, tparam) =>
               // note: can't use args1 in selector, because Binds got replaced
               val asym = arg.symbol
-              def abounds = asym.info.bounds
-              def tbounds = tparam.info.bounds
               def enhanceBounds(): Unit = {
-                val TypeBounds(lo0, hi0) = abounds
-                val TypeBounds(lo1, hi1) = tbounds.subst(tparams, argtypes)
+                val lo0 = asym.info.lowerBound
+                val hi0 = asym.info.upperBound
+                val lo1 = tparam.info.lowerBound.subst(tparams, argtypes)
+                val hi1 = tparam.info.upperBound.subst(tparams, argtypes)
                 val lo = lub(List(lo0, lo1))
                 val hi = glb(List(hi0, hi1))
-                if (!(lo =:= lo0 && hi =:= hi0))
+                if (!(lo =:= lo0 && hi =:= hi0)){
+                  val abounds = TypeBounds(lo0, hi0)
```

Comment:
This is only used for logging now, right? Can you make it not be evaluated when `Ylog` is off?

Reply:
Sure!